server/rules.go

```diff
@@ -0,0 +1,358 @@
+// Copyright 2020 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/RedHatInsights/insights-operator-utils/responses"
+	"github.com/RedHatInsights/insights-results-aggregator/storage"
+	"github.com/RedHatInsights/insights-results-aggregator/types"
+)
+
+// getRule returns rule with content for provided rule ID and rule error key
+func (server *HTTPServer) getRule(writer http.ResponseWriter, request *http.Request) {
+	ruleID, err := readRuleID(writer, request)
+	if err != nil {
+		// everything has been handled already
+		return
+	}
+
+	errorKey, err := readErrorKey(writer, request)
+	if err != nil {
+		// everything has been handled already
+		return
+	}
+
+	ruleWithContent, err := server.Storage.GetRuleWithContent(ruleID, errorKey)
+	if err != nil {
+		handleServerError(writer, err)
+		return
+	}
+
+	err = responses.SendOK(writer, responses.BuildOkResponseWithData("rule", ruleWithContent))
+	if err != nil {
+		log.Error().Err(err).Msg(responseDataError)
+	}
+}
+
+// likeRule likes the rule for current user
+func (server *HTTPServer) likeRule(writer http.ResponseWriter, request *http.Request) {
```

Comment:
Still think it will be nice to move `vote` functions to separate file.

Reply:
done https://github.com/RedHatInsights/insights-results-aggregator/pull/819/commits/941e1e4fe2e3f5843eb43176a35d786c2f5ce86d